src/rules/color-named/index.js

```diff
@@ -55,6 +56,11 @@ export default function (expectation, options) {
       valueParser(decl.value).walk(node => {
         const { value, type, sourceIndex } = node
 
+        if (
+          optionsMatches(options, "ignore", "inside-function")
+          && type === "function"
+          && !colorFunctionNames.has(value.toLowerCase())) { return false }
```

Comment:
This is a check for a function that is a color representation e.g. `rgb`, `gray`....See @jeddy3's comments above.